### DISCUSSION TITLE
Passwordless cockpit access for development VMs

### DISCUSSION THREAD

**Discussion by mhitza** [cockpit-project/cockpit#16758] [Q&A] [answered]
I'm setting up a vagrant box and I would like to expose access to cockpit without password authentication (which among other annoyances, requires an additional step of setting up a password for the vagrant user). I couldn't find an easy way to do this, and not sure how/if it's possible to repurpose cockpit-desktop in order to achieve this (I'd prefer the setup to be fully contained within the VM in order to make the VM usable on any OS).

**Comment by martinpitt** [comment, marked as answer]
It's possible to remove both the https layer (to avoid unknown certificates) and authentication with changing the systemd unit -- `systemctl edit cockpit` or creating /etc/systemd/system/cockpit.service.d/override.conf some other way:

```ini
Service]
ExecStartPre=
ExecStart=
ExecStart=/usr/libexec/cockpit-ws --no-tls --local-session=/usr/bin/cockpit-bridge
User=root
```

That works, but is admittedly not that easy to discover :wink: 

Note that this requires to disable SELinux (`setenforce 0` or permanently in /etc/selinux/config)

**Reply by geoidesic** [reply]
Not a good solution if you have to disable the very thing that cockpit is designed to monitor???
